src/bot.ts

```diff
@@ -0,0 +1,107 @@
+import { Issue, IssueComment } from "@octokit/webhooks-types";
+
+import { CommentsApi } from "./github/comments";
+import { Merger } from "./github/merger";
+import { ActionLogger } from "./github/types";
+
+const BOT_COMMAND = "/bot";
+
+type Command = "merge" | "cancel" | "help";
+
+const botCommands = `
+**Available commands**
+
+- \`/bot merge\`: Enables auto-merge for Pull Request
+- \`/bot cancel\`: Cancels auto-merge for Pull Request
+- \`/bot help\`: Shows this menu
+
+For more information see the [documentation](https://github.com/paritytech/auto-merge-bot)
+`;
+
+export class Bot {
+  constructor(
+    private readonly comment: IssueComment,
+    private readonly pr: Issue,
+    private readonly logger: ActionLogger,
+    private readonly commentsApi: CommentsApi,
+  ) {}
+
+  /** Verifies if the author is the also the author of the PR or a member of the org */
+  async canTriggerBot(): Promise<boolean> {
+    this.logger.debug("Evaluating if user can trigger the bot");
+    const author = this.pr.user.id;
+    if (this.comment.user.id === author) {
+      this.logger.debug("Author of comment is also author of PR");
+      return true;
+    }
+    this.logger.debug("Author of comment is not the author of the PR");
+
+    return await this.commentsApi.userBelongsToOrg(this.comment.user.login);
+  }
+
+  async run(merger: Merger): Promise<void> {
+    this.logger.info("Running action on comment: " + this.comment.html_url);
+    if (!this.comment.body.startsWith(BOT_COMMAND)) {
+      this.logger.info(
+        `Ignoring comment ${this.comment.html_url} as it does not start with '${BOT_COMMAND}'`,
+      );
+      return;
+    }
+
+    if (this.pr.state === "closed") {
```

Comment:
should we check against draft too?

Reply:
The state object only has `closed` or `open`.

I would say no anyways because it still needs to have the reviews. You can always set a project to auto-merge even if it is in draft state.

What do you think?